src/main/java/org/gridsuite/filter/server/repositories/proxies/expertfiler/ExpertFilterRepositoryProxy.java

```diff
@@ -148,15 +187,43 @@ public ExpertFilterEntity fromDto(AbstractFilter dto) {
     }
 
     public static ExpertRuleEntity dtoToEntity(AbstractExpertRule filter) {
-        var expertFilterEntityBuilder = ExpertRuleEntity.builder()
-                .id(UUID.randomUUID())
-                .combinator(filter.getCombinator())
-                .operator(filter.getOperator())
-                .dataType(filter.getDataType())
-                .field(filter.getField())
-                .value(filter.getStringValue());
-        expertFilterEntityBuilder.rules(dtoToEntities(filter.getRules(), expertFilterEntityBuilder.build()));
-        return expertFilterEntityBuilder.build();
+        ExpertRuleEntity.ExpertRuleEntityBuilder<?, ?> expertRuleEntityBuilder = null;
+        if (filter.getDataType() == DataType.COMBINATOR) {
+            expertRuleEntityBuilder = ExpertRuleEntity.builder()
+                    .id(UUID.randomUUID())
+                    .combinator(filter.getCombinator())
+                    .operator(filter.getOperator())
+                    .dataType(filter.getDataType())
+                    .field(filter.getField());
+        }
+        if (filter.getDataType() == DataType.PROPERTIES) {
+            PropertiesExpertRule propertiesRule = (PropertiesExpertRule) filter;
+            expertRuleEntityBuilder = ExpertRulePropertiesEntity.builder()
+                    .id(UUID.randomUUID())
+                    .combinator(filter.getCombinator())
+                    .operator(filter.getOperator())
+                    .dataType(filter.getDataType())
+                    .field(filter.getField())
+                    .propertyValues(propertiesRule.getPropertyValues())
+                    .propertyName(propertiesRule.getPropertyName());
+
+        } else if (filter.getDataType() == DataType.BOOLEAN || filter.getDataType() == DataType.NUMBER || filter.getDataType() == DataType.STRING || filter.getDataType() == DataType.ENUM || filter.getDataType() == DataType.FILTER_UUID) {
+            expertRuleEntityBuilder = ExpertRuleValueEntity.builder()
+                    .id(UUID.randomUUID())
+                    .combinator(filter.getCombinator())
+                    .operator(filter.getOperator())
+                    .dataType(filter.getDataType())
+                    .field(filter.getField())
+                    .value(filter.getStringValue());
+        }
+        if (expertRuleEntityBuilder == null) {
+            throw new PowsyblException("Unsupported data type: " + filter.getDataType());
+        }
```

Comment:
In my opinion, we can extract to a method: getBuilder(rule) which return a builder in order to reuse in dtoToEntity and in the map() in dtoToEntities.

    public static ExpertRuleEntity.ExpertRuleEntityBuilder<?, ?> getBuilder(AbstractExpertRule filter) {
        ExpertRuleEntity.ExpertRuleEntityBuilder<?, ?> expertRuleEntityBuilder = null;
        if (filter.getDataType() == DataType.COMBINATOR) {
            expertRuleEntityBuilder = ExpertRuleEntity.builder()
                    .id(UUID.randomUUID())
                    .combinator(filter.getCombinator())
                    .operator(filter.getOperator())
                    .dataType(filter.getDataType())
                    .field(filter.getField());
        }
        if (filter.getDataType() == DataType.PROPERTIES) {
            PropertiesExpertRule propertiesRule = (PropertiesExpertRule) filter;
            expertRuleEntityBuilder = ExpertRulePropertiesEntity.builder()
                    .id(UUID.randomUUID())
                    .combinator(filter.getCombinator())
                    .operator(filter.getOperator())
                    .dataType(filter.getDataType())
                    .field(filter.getField())
                    .propertyValues(propertiesRule.getPropertyValues())
                    .propertyName(propertiesRule.getPropertyName());

        } else if (filter.getDataType() == DataType.BOOLEAN || filter.getDataType() == DataType.NUMBER || filter.getDataType() == DataType.STRING || filter.getDataType() == DataType.ENUM || filter.getDataType() == DataType.FILTER_UUID) {
            expertRuleEntityBuilder = ExpertRuleValueEntity.builder()
                    .id(UUID.randomUUID())
                    .combinator(filter.getCombinator())
                    .operator(filter.getOperator())
                    .dataType(filter.getDataType())
                    .field(filter.getField())
                    .value(filter.getStringValue());
        }
        if (expertRuleEntityBuilder == null) {
            throw new PowsyblException("Unsupported data type: " + filter.getDataType());
        }

        return expertRuleEntityBuilder;
    }

That can avoid duplication of code

Reply:
done